src/Microsoft.AspNetCore.SystemWebAdapters/Build/NuGetContentFiles.targets

```diff
@@ -0,0 +1,53 @@
+<Project>
+
+  <!--
+      Add NuGet content to the build output if specified:
+      <PackageReference Include="..." CopyContent="true" ... />
+       -->
+  <Target Name="AddNugetContentFiles" BeforeTargets="AssignTargetPaths">
+
+    <!-- Identity PackageReference and paths to packages -->
+    <ItemGroup>
+      <_PackageReferenceProperties Include="Pkg$([System.String]::Copy('%(PackageReference.Identity)').Replace('.','_'))">
+        <PackageName>%(PackageReference.Identity)</PackageName>
+        <CopyContent>%(PackageReference.CopyContent)</CopyContent>
+      </_PackageReferenceProperties>
+      <_PackageReferenceProperties Update="@(_PackageReferenceProperties)" PackageRootPath="$(%(_PackageReferenceProperties.Identity))" />
+      <_PackageReferenceCopyContent Include="@(_PackageReferenceProperties)" Condition="'%(CopyContent)' == 'true'" />
+    </ItemGroup>
+
+    <!-- Check for tools files -->
+    <ItemGroup>
+      <_ToolsFiles Include="%(_PackageReferenceCopyContent.PackageRootPath)/tools/*.ps1">
+        <PackageName>%(PackageName)</PackageName>
+      </_ToolsFiles>
+      <_ToolsPackage Include="%(_ToolsFiles.PackageName)" />
+    </ItemGroup>
+
+    <!-- Get content files-->
+    <ItemGroup>
+      <_ContentPkgProperties Include="@(_PackageReferenceCopyContent)" />
+
+      <!-- Filter out packages that need GeneratePathProperty -->
+      <_ValidContentPkgProperties Include="@(_ContentPkgProperties)" Condition=" '%(PackageRootPath)' != '' " />
+      <_InvalidContentPkgProperties Include="@(_ContentPkgProperties)" Condition=" '%(PackageRootPath)' == '' " />
+    </ItemGroup>
+
+    <Warning
+      Condition=" '@(_InvalidContentPkgProperties)' != '' "
+      Text="%(_InvalidContentPkgProperties.PackageName) does not contain a tools directory and thus needs 'GeneratePathProperty' == 'true'." />
+
+    <ItemGroup Condition=" '@(_ValidContentPkgProperties)' != '' ">
+      <_NuGetContentFiles Include="%(_ValidContentPkgProperties.PackageRootPath)\Content\**\*">
+        <BasePath>%(_ValidContentPkgProperties.PackageRootPath)\Content</BasePath>
+        <PackageIdentity>%(_ValidContentPkgProperties.PackageName)</PackageIdentity>
+      </_NuGetContentFiles>
+
+      <Content Include="@(_NuGetContentFiles)">
```

Comment:
Some users hate that these content files show up in Solution Explorer and want this item to have `<Visible>false</Visible>`.

https://github.com/NuGet/Home/issues/4856

By default, NuGet shows them but I don't like that.  Just a heads up that people might ask for these to be hidden.

Reply:
They currently don't show up solution explorer. I'm assuming that's because I have set the target `BeforeTargets="AssignTargetPaths"`.  Are there conditions in which it will show up I just haven't seen?

Reply:
Oh sorry I thought I scrolled up and didn't see this in a target.  Only `Content` and `None` items outside of targets show up.